libqtile/config.py

```diff
@@ -73,12 +75,14 @@ class KeyChord:
     key:
         A key specification, e.g. "a", "Tab", "Return", "space".
     submappings:
-        A list of Key declarations to bind in this chord
+        A list of Key or KeyChord declarations to bind in this chord.
     mode:
-        A string with vim like mode name if it's set chord not end
-        after use one of submapings or Esc key
+        A string with vim like mode name. If it's set, the chord mode will
+        not be left after a keystroke (except for Esc which always leaves the
+        current chord/mode).
     """
-    def __init__(self, modifiers: List[str], key: str, submapings: List[Key], mode: str = ""):
+    def __init__(self, modifiers: List[str], key: str,
+                 submapings: List[Union[Key, KeyChord]], mode: str = ""):
```

Comment:
Forgive me as I am not very knowledgable about Python's type checking (something I avoided for the longest time...). I read through PEP 563 but it still isn't clear to me why it is needed for this use case. Why wouldn't the change you have made to this annotation work without the added futures import?

An unrelated comment -- what's up with the `submapings` spelling? I assume this is an error that predates this PR. Should we fix that?

Reply:
And as I move on to the next commit I see you have fixed the spelling! Ignore that comment :D

Reply:
So before PEP 563 or whatever, I mentioned in the commit message, using a class inside itself as a type annotation was only possible with a really ugly hack:
```py
class MyClass:
    pass

class MyClass:
    ...
    def myFunc(self, other: MyClass):
         ...
```
because while defining `MyClass`, the `MyClass` doesn't yet exist. With that PEP / future import we can enable lazy type annotation evaluation so that we don't get a `NameError` or what this thing actually throws without that import.
From 3.10 onwards this behaviour will be default, so the import can be removing when python 3.9 support will be eventually dropped in the distant future...

Reply:
Behaviour in python 3.9.2:
```py
~ via 🐍 v3.9.2 took 16s 
❯ python
>>> class MyClass:
...     ...
...     def myFunc(self, other: MyClass):
...          ...
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 3, in MyClass
NameError: name 'MyClass' is not defined
>>> ^D
~ via 🐍 v3.9.2 took 37s 
❯ python
>>> from __future__ import annotations  # with this import everything works fine
>>> class MyClass:
...     ...
...     def myFunc(self, other: MyClass):
...          ...
... 
>>> 
```

Reply:
OK, awesome, thanks a lot for the explanation! All good :)